test/integration/targets/cloudwatchlogs/playbooks/roles/cloudwatchlogs_log_group_metric_filter/tasks/main.yml

```diff
@@ -0,0 +1,151 @@
+        filter_name: MadeWithAnsible
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: fluentd_metrics
+          metric_value: "$.value"
+      register: out
+
+    - name: create metric filter
+      assert:
+        that:
+          - out is changed
+
+    - name: re-set metric filter on ansible/integrationtest
+      cloudwatchlogs_log_group_metric_filter:
+        <<: *aws_connection_info
+        log_group_name: ansible/integrationtest
+        filter_name: MadeWithAnsible
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: fluentd_metrics
+          metric_value: "$.value"
+      register: out
+
+    - name: metric filter must not change
+      assert:
+        that:
+          - out is not changed
+
+    - name: update matric transformation on ansible/integrationtest
+      cloudwatchlogs_log_group_metric_filter:
+        <<: *aws_connection_info
+        log_group_name: ansible/integrationtest
+        filter_name: MadeWithAnsible
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: made_with_ansible
+          metric_value: "$.value"
+      register: out
+
+    - name: update metric filter
+      assert:
+        that:
+          - out is changed
+
+    - name: update filter_pattern on ansible/integrationtest
+      cloudwatchlogs_log_group_metric_filter:
+        <<: *aws_connection_info
+        log_group_name: ansible/integrationtest
+        filter_name: MadeWithAnsible
+        filter_pattern: '{ ($.value = *) && ($.hostname = "ansible")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: made_with_ansible
+          metric_value: "$.value"
+      register: out
+
+    - name: update metric filter
+      assert:
+        that:
+          - out is changed
+
+    - name: checkmode delete metric filter on ansible/integrationtest
+      cloudwatchlogs_log_group_metric_filter:
+        <<: *aws_connection_info
+        log_group_name: ansible/integrationtest
+        filter_name: MadeWithAnsible
+        state: absent
+      check_mode: yes
+      register: out
+
+    - name: check_mode state must be changed
+      assert:
+        that:
+          - out is changed
+
+    - name: delete metric filter on ansible/integrationtest
+      cloudwatchlogs_log_group_metric_filter:
+        <<: *aws_connection_info
+        log_group_name: ansible/integrationtest
+        filter_name: MadeWithAnsible
+        state: absent
+      register: out
+
+    - name: delete metric filter
+      assert:
+        that:
+          - out is changed
```

Comment:
We should also have assertions for the length of `out.metric_filters` in each test, so that here we can assert it is 0.